test/parallel/test-https-agent-abort-controller.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const https = require('https');
+const Agent = https.Agent;
+const fixtures = require('../common/fixtures');
+
+const { getEventListeners } = require('events');
+const agent = new Agent();
+
+const options = {
+  key: fixtures.readKey('agent1-key.pem'),
+  cert: fixtures.readKey('agent1-cert.pem')
+};
+
+const server = https.createServer(options);
+
+server.listen(0, common.mustCall(async () => {
+  const port = server.address().port;
+  const host = 'localhost';
+  const options = {
+    port: port,
+    host: host,
+    rejectUnauthorized: false,
+    _agentKey: agent.getName({ port, host })
+  };
+
+  function postCreateConnection() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const connection = agent.createConnection({ ...options, signal });
+      connection.on('error', common.mustCall((err) => {
```

Comment:
I'd use `events.once` in these (instead of new Promise) with an `async function`